scylla/src/transport/topology.rs

```diff
@@ -301,12 +400,276 @@ async fn query_keyspaces(conn: &Connection) -> Result<HashMap<String, Keyspace>,
 
         let strategy: Strategy = strategy_from_string_map(strategy_map)?;
 
-        result.insert(keyspace_name, Keyspace { strategy });
+        result.insert(
+            keyspace_name.clone(),
```

Comment:
Tip: You can avoid cloning here, too, see my other comment about `(keyspace_name, table_name)`.